docs/source/essentials/queries.md

```diff
@@ -0,0 +1,280 @@
+            Log.i(TAG, response.toString());
+          }
+
+          @Override public void onFailure(@NotNull ApolloException e) {
+            Log.e(TAG, e.getMessage(), e);
+          }
+        }, uiHandler));
+```
+
+> By default, Apollo will deliver query results on a background, so you have to provide a handler in enqueue if you're using the result to update the UI.
+
+The `ApolloCall.Callback` also provides error handling methods for request parsing failed, network error and request cancelled, amongst others.
+
+In addition to the `data` property, `response` contains an `errors` list with GraphQL errors (for more on this, see the sections on [error handling](https://facebook.github.io/graphql/#sec-Error-handling) and the [response format](https://facebook.github.io/graphql/#sec-Response-Format) in the GraphQL specification).
+
+<h2 id="typed-query-results">Typed query results</h2>
+
+Query results are defined as nested immutable classes that at each level only contain the properties defined in the corresponding part of the query definition. 
+This means the type system won't allow you to access fields that are not actually fetched by the query, even if they *are* part of the schema.
+
+For example, given the following schema:
+
+```graphql
+enum Episode { NEWHOPE, EMPIRE, JEDI }
+
+interface Character {
+  id: String!
+  name: String!
+  friends: [Character]
+  appearsIn: [Episode]!
+ }
+
+ type Human implements Character {
+   id: String!
+   name: String!
+   friends: [Character]
+   appearsIn: [Episode]!
+   height(unit: LengthUnit = METER): Float
+ }
+
+ type Droid implements Character {
+   id: String!
+   name: String!
+   friends: [Character]
+   appearsIn: [Episode]!
+   primaryFunction: String
+}
+```
+
+And the following query:
+
+```graphql
+query HeroAndFriendsNames($episode: Episode) {
+  hero(episode: $episode) {
+    name
+    friends {
+      name
+    }
+  }
+}
+```
+
+You can fetch results and access data using the following code:
+
+```java
+final HeroAndFriendsNames heroAndFriendsQuery = HeroAndFriendsNames.builder()
+    .episode(NEWHOPE)
+    .build();
+
+apolloClient().query(heroAndFriendsQuery)
+    .enqueue(new ApolloCallback<>(new ApolloCall.Callback<HeroAndFriendsNames.Data>() {
+      @Override public void onResponse(@NotNull Response<HeroAndFriendsNames.Data> response) {
+        Log.i(TAG, response.toString());
+      }
+
+      @Override public void onFailure(@NotNull ApolloException e) {
+        Log.e(TAG, e.getMessage(), e);
+      }
+    }, uiHandler));
+}
+```
+
+Because the above query won't fetch `appearsIn`, this property is not part of the returned result type and cannot be accessed here.
+
+<h2 id="cache-policy">Support For Cached Responses</h2>
+
+As explained in more detail in [the section on watching queries](watching-queries.html), Apollo-android client allows you to keep a client-side cache of query results, making it suitable for use even while offline.
```

Comment:
I would remove anything about caching and make caching its own document

Reply:
Done. Will create a new PR after this.